includes/acl/core/memory_utils.h

```diff
@@ -167,19 +167,34 @@ namespace acl
 		template<typename Type>
 		struct safe_underlying_type<Type, false> { using type = Type; };
 
+		template<typename DstType, typename SrcType, bool isreal = false>
+		struct is_float_impl {
+			static bool is_static_cast_safe_float(SrcType input)
```

Comment:
remove the `is_static_cast_safe` function since it's only called in a single place and move your c++17 comment and the next line into the caller. Rename this struct into `is_static_cast_safe` and for consistency, the function inside should be called `test()` to match the `std` convention for similar code